book/src/client.md

```diff
@@ -157,7 +157,54 @@ Signals are like methods, except they don't expect a reply. They are typically e
 to notify interested peers of any changes to the state of the service. zbus provides you a
 [`Stream`]-based API for receiving signals.
 
-Let's look at this API in action, with an example where we get our location from
+Let's look at this API in action, with an example where we monitor started systemd units.
+
+```rust,no_run
+# // NOTE: When changing this, please also keep `zbus/examples/watch-systemd-jobs.rs` in sync.
+use async_std::stream::StreamExt;
+use zbus::Connection;
+use zbus_macros::dbus_proxy;
+use zvariant::OwnedObjectPath;
+
+# fn main() {
+#     async_io::block_on(watch_systemd_jobs()).expect("Error listening to signal");
+# }
+
+#[dbus_proxy(
+    default_service = "org.freedesktop.systemd1",
+    default_path = "/org/freedesktop/systemd1",
+    interface = "org.freedesktop.systemd1.Manager"
+)]
+trait Systemd1Manager {
+    // Defines signature for D-Bus signal named `JobNew`
+    #[dbus_proxy(signal)]
+    fn job_new(&self, id: u32, job: OwnedObjectPath, unit: String) -> zbus::Result<()>;
+}
+
+async fn watch_systemd_jobs() -> zbus::Result<()> {
+    let connection = Connection::system().await?;
+    // `Systemd1ManagerProxy` is generated from `Systemd1Manager` trait
+    let systemd_proxy = Systemd1ManagerProxy::new(&connection).await?;
+    // Method `receive_job_new` is generated from `job_new` signal
+    let mut new_jobs_stream = systemd_proxy.receive_job_new().await?;
+
+    while let Some(msg) = new_jobs_stream.next().await {
+        // struct `JobNewArgs` is generated from `job_new` signal function arguments
+        let body: JobNewArgs = msg.args().expect("Error parsing message");
```

Comment:
nit: `args` is a better name IMO and consistent with a lot of code we've in zbus, including the geoclue example.